### PR TITLE
Updated README.md and Modified Workflow Trigger Events

### LINKED ISSUE #21: README.md
The README should have information on how one can:

- [x] setup a local dev environment (venv + pip)
- [x] launch the API (uvicorn ... + set environment variables)
- [x] how to run the pytest tests (env variables are set + the server is reachable?)

### LINKED ISSUE #42: Only run test suite in pull requests
Our excellent test suite currently [runs on two events](https://github.com/neurobagel/bagelapi/blob/main/.github/workflows/test.yaml): push and pull_request ([see also](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows)). I think they should only run for pull _request so we don't run them twice during a PR.

### PR DESCRIPTION
**Pull request by rmanaem**
Closes #21 
Closes #42

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,98 @@
+[//]: <> (Do we need to include a section explaining the server connection whether that'd be McGill vpn or something else?)
```
Comment (surchs) [inline README.md:1]: is this leftover from our discussion?

File: app/api/models.py [hunk 2]
```diff
@@ -1,10 +1,9 @@
 """Data models."""
 
-from typing import Literal
-
 from fastapi import Query
 from fastapi.exceptions import HTTPException
 from pydantic import BaseModel, constr, root_validator
+from typing_extensions import Literal
```
Comment (surchs) [inline app/api/models.py:6]: this may be a good idea (I don't know), but let's handle it in a different PR so we don't confuse ourselves later :)

**Review by surchs — CHANGES_REQUESTED** [review]
Thanks a lot for this PR, @rmanaem :tada:! Very nice README file! I flagged two things that seem to have accidentally snuck into the PR, but otherwise good to go!

File: README.md (hunk 1 above)
Comment (rmanaem) [reply README.md:1]: Yea I left this as a reminder for the discussion.

File: app/api/models.py (hunk 2 above)
Comment (rmanaem) [reply app/api/models.py:6]: It is in a way related to the README.md file update. While I was adding badges, I updated the test suite to run for Python 3.7-3.10. The suite for 3.7 failed due to a discrepancy in the Literal package import statement.

Comment (surchs) [reply app/api/models.py:6]: ah, great thinking! That's good to check. But also a big enough thing that we should make a separate issue to discuss what versions we want to support and then be intentional about that. So I'd remove it here and bring it back when we address that.

File: README.md (hunk 1 above)
Comment (surchs) [reply README.md:1]: OK. But we wrapped up the discussion and said we wouldn't include it for now, no? So I'd remove it here so we don't wonder the same thing every time we see the raw text.

Comment (rmanaem) [reply README.md:1]: I think we decided to have a discussion in PR(or draft) instead but anyways I had this there before bringing it up in the issue. I'll take care of it.

**Review by surchs — APPROVED** [review]
looks great! Thanks @rmanaem :tada: gtg

**Pull request merged**